source/_integrations/youtube.markdown

```diff
@@ -0,0 +1,68 @@
+---
+title: YouTube
+description: Instructions on how to integrate YouTube within Home Assistant.
+ha_category:
+  - Sensor
+ha_release: '2023.6'
+ha_iot_class: Cloud Polling
+ha_config_flow: true
+ha_codeowners:
+  - '@joostlek'
+ha_domain: youtube
+ha_platforms:
+  - sensor
+ha_integration_type: service
+---
+
+The YouTube integration allows you to connect YouTube channels to Home Assistant.
+
+## Prerequisites
+
+You need to configure developer credentials to allow Home Assistant to access your Google Account.
+These credentials are the same as the ones for [Nest](/integrations/nest) and [Google Sheets](/integrations/google_sheets).
+These are not the same as the one for [Google Calendar](/integrations/google).
+If you have already set up the correct credentials, you can do step 1 and then skip to step 13 on the below instructions.
+
+{% details "Generate Client ID and Client Secret" %}
+
+This section explains how to generate a Client ID and Client Secret on
+[Google Developers Console](https://console.cloud.google.com/apis/library/youtube.googleapis.com).
+
+1. First, go to the Google Developers Console to enable the [YouTube Data API v3](https://console.cloud.google.com/apis/library/youtube.googleapis.com?project=home-assistant-17698)
+2. The wizard will ask you to choose a project to manage your application. Select a project and click continue.
+3. Verify that your YouTube Data API v3 was enabled and click 'Go to credentials'
+4. Navigate to APIs & Services (left sidebar) > [Credentials](https://console.cloud.google.com/apis/credentials)
+5. Click on the field on the left of the screen, **OAuth Consent Screen**.
+6. Select **External** and **Create**.
+7. Set the *App Name* (the name of the application asking for consent) to anything you want, e.g., *Home Assistant*.
+8. You then need to select a *Support email*. To do this, click the drop-down box and select your email address.
+9. You finally need to complete the section: *Developer contact information*. To do this, enter your email address (the same as above is fine).
+10. Scroll to the bottom and click **Save and Continue**. You don't have to fill out anything else, or it may enable additional review.
+11. You will then be automatically taken to the Scopes page. You do not need to add any scopes here, so click Save and Continue to move to the Optional info page. You do not need to add anything to the Optional info page, so click Save and Continue, which will take you to the Summary page. Click Back to Dashboard.
+12. Click **OAuth consent screen** again and set *Publish Status* to **Production** otherwise your credentials will expire every 7 days.
```

Comment:
```suggestion
12. Select **OAuth consent screen** again and set *Publish Status* to **Production**. Otherwise your credentials will expire every 7 days.
```